File: LeetCode_Solutions/49-group-anagrams/group-anagrams.java
class Solution {
  public List<List<String>> groupAnagrams(String[] strs) {
    Map<String, List<String>> keyToAnagrams = new HashMap<>();

    for (final String str : strs) {
      char[] chars = str.toCharArray();
      Arrays.sort(chars);
      String key = String.valueOf(chars);
      //System.out.println(key);
      keyToAnagrams.computeIfAbsent(key, k -> new ArrayList<>()).add(str);
      //System.out.println(keyToAnagrams);
    }

    return new ArrayList<>(keyToAnagrams.values());
  }
}

File: LeetCode_Solutions/49-group-anagrams/group-anagrams.py
class Solution:
  def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
    dict = collections.defaultdict(list)
    #print(dict)
    # for key,values in dict.items():
        #  print(f'{key}: {values}')
    for str in strs:
      key = ''.join(sorted(str))
      dict[key].append(str)

    return dict.values()